lib/utils/analytics/generatePayload.js

```diff
@@ -98,12 +98,15 @@ module.exports = async (serverless) => {
         region: isAwsProvider ? provider.getRegion() : providerConfig.region,
       },
       plugins: serviceConfig.plugins ? serviceConfig.plugins.modules || serviceConfig.plugins : [],
-      functions: Object.values(serviceConfig.functions).map((functionConfig) => ({
-        runtime: functionConfig.runtime || defaultRuntime,
-        events: functionConfig.events.map((eventConfig) => ({
-          type: Object.keys(eventConfig)[0] || null,
-        })),
-      })),
+      functions: Object.values(serviceConfig.functions).map((functionConfig) => {
+        const functionEvents = functionConfig.events || [];
```

Comment:
Why this change is necessary? Normally `functionConfig.events` is guaranteed internally

Reply:
It was due to a quirky test setup of https://github.com/serverless/serverless/blob/bf83f3c406b576599c7f37a37e583ac6a530b1d3/test/unit/lib/utils/analytics/generatePayload.test.js#L18
In this case, the existence of `events` as an array is not ensured on `functions`. I've decided to patch it in a way that will ensure that `generatePayload` will default to an empty list in such edge cases.

Reply:
hmm.. but it was more an issue in _test_ than in tested module. 

Technically `serverless` instance guarantees that that every function has `events` array, and in such case we should not introduce a noise with such checks (same as we assume that all values match configuration schema)

Reply:
However, as I think of it. In future we may loose that guarantee. Especially when we will pass  a _configuration input_ directly (by schema, `events` is not a required property).

So after all I think it's fine to add such check here